hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/AbstractBatchEntitySelectFetchInitializer.java

```diff
@@ -89,12 +91,32 @@ protected void resolveKey(
 		}
 		else {
 			entityKey = new EntityKey( entityIdentifier, concreteDescriptor );
-
 			state = State.KEY_RESOLVED;
+			registerResolutionListener();
+		}
+	}
 
+	protected Object getExistingInstance(RowProcessingState rowProcessingState) {
+		if ( entityKey != null ) {
+			final SharedSessionContractImplementor session = rowProcessingState.getSession();
+			final PersistenceContext persistenceContext = session.getPersistenceContext();
+			final Object instance = persistenceContext.getEntity( entityKey );
```

Comment:
Could `instance` be an uninitialized proxy in which case we would have to return `null` here to make sure batch loading still happens? If so, we should IMO also rename the method to `getExistingInitializedInstance` to clarify this.